engine/src/execution_graph/logic_controllers/CacheMachine.cpp

```diff
@@ -479,14 +480,25 @@ std::unique_ptr<ral::cache::CacheData>  CacheMachine::get_or_wait_CacheData(size
 }
 
 std::unique_ptr<ral::frame::BlazingTable> CacheMachine::pullFromCache() {
+    static int global_index = -1;
```

Comment:
This cant be static. This has to be a member of the CacheMachine. 
Also, the same paradigm has to be applied to `pullCacheData()`

Reply:
Also for `pullUnorderedFromCache()` make it so that if is_array_access == true, then just use `pullFromCache()`